scipy/special/_basic.py

```diff
@@ -3156,3 +3157,46 @@ def zeta(x, q=None, out=None):
         return _ufuncs._riemann_zeta(x, out)
     else:
         return _ufuncs._zeta(x, q, out)
+
+      
+def softplus(x, **kwargs):
+    r"""
+    Compute the softplus function element-wise.
+
+    The softplus function is defined as: ``softplus(x) = log(1 + exp(x))``.
+    It is a smooth approximation of the rectifier function (ReLU).
+
+    Parameters
+    ----------
+    x : array_like
+        Input array-like object.
+
+    **kwargs :
+        out: Specify the output array.
+        where: Boolean array to control ufunc calculation at each position.
+        axes: List of tuples specifying indices of axes for generalized ufuncs.
+        axis: Single axis over which a generalized ufunc should operate.
+        keepdims: If True, keep reduced axes as dimensions with size one.
+        casting: Policy for permitted casting.
+        order: Iteration order/memory layout of the output array.
+        dtype: Override the output array's dtype.
+        subok: If False, the output will always be a strict array.
+        signature: Dtype(s), tuple of DTypes, or special signature string indicating input/output types of a ufunc.
+        extobj: List of length 3 specifying ufunc buffer-size, error mode integer, and error callback function.
+
+    Returns
+    -------
+    softplus : array_like
```

Comment:
The return type is `ndarray` (or a scalar, but we document it as `ndarray` - you can copy how the `np.logaddexp` return value is documented).